### PR TITLE
Adding additional fixtures for preferred-citations

### PR DESCRIPTION
**Pull request by arfon**
👋 @hainesr – adding a few additional fixtures for the `preferred-citation` key. Note, there are failing tests currently – in testing out the most recent changes I found that the APA and BibTeX outputs were missing some of the key fields for `@article` type entries.

### CODE REVIEW

File: test/converted/smith-et-al.apa [hunk 1]
```diff
@@ -0,0 +1 @@
+Smith, A. M., Katz, D. S., Niemeyer, K. E., & FORCE11 Software Citation Working Group (2016). Software citation principles. PeerJ Computer Science, 2(123), e86. https://doi.org/10.7717/peerj-cs.86
```
Comment (arfon) [inline test/converted/smith-et-al.apa:1]: This is how I *think* the APA formatted citation should look for v7 of APA.

File: test/converted/tidyverse-joss-paper.apa [hunk 2]
```diff
@@ -0,0 +1 @@
+Wickham, H. (2019). Welcome to the Tidyverse. Journal of Open Source Software, 4(43), 1686. https://doi.org/10.21105/joss.01686
```
Comment (arfon) [inline test/converted/tidyverse-joss-paper.apa:1]: Again, this is how I believe an APA 7 formatted article citation should look.

File: test/files/tidyverse-software.cff [hunk 3]
```diff
@@ -0,0 +1,13 @@
+# Note this is a made up entry.
```
Comment (arfon) [inline test/files/tidyverse-software.cff:1]: Making up a minimal entry for the Tidyverse software here.

File: test/files/tidyverse-joss-paper.cff [hunk 4]
```diff
@@ -0,0 +1,29 @@
+# Note this is a made up entry.
```
Comment (arfon) [inline test/files/tidyverse-joss-paper.cff:1]: Imaginary Tidyverse CFF where the authors want the JOSS paper to be cited.

File: test/files/smith-et-al.cff [hunk 5]
```diff
@@ -0,0 +1,20 @@
+cff-version: 1.2.0
```
Comment (arfon) [inline test/files/smith-et-al.cff:1]: A minimal CFF file where there is only a preferred-citation block

Comment (arfon) [reply test/files/smith-et-al.cff:1]: Not sure if this would be valid?

**Comment by arfon** [comment]
Note, I've not tried to debug this yet, rather, thought it would be useful to start with some breaking tests 😬

**Comment by hainesr** [comment]
Thanks for this @arfon. Just to note, I am working on this branch directly now to get these tests passing, so that the changes will also show up here. I may rebase with extreme prejudice 😆 !

File: test/converted/smith-et-al.apa (hunk 1 above)
Comment (hainesr) [reply test/converted/smith-et-al.apa]: The other APA examples we have, have a `.` after the authors so I've added that in the hope it's correct 😬

File: test/files/smith-et-al.cff (hunk 5 above)
Comment (hainesr) [reply test/files/smith-et-al.cff:1]: It's not valid, but the formatters cope with it thanks to all the required info being in `preferred-citation`.

**Review by hainesr — APPROVED** [review]
This looks good to me!

**Comment by hainesr** [comment]
Hi @arfon, please could you give this a once-over, and if it looks good I'll get it merged.

Thanks for adding the new fixtures; we definitely had some serious gaps in our tests/functionality.

File: test/files/smith-et-al.cff [hunk 6]
```diff
@@ -1,5 +1,14 @@
...
 message: If you use this software, please cite it as below.
+authors:
+  - family-names: Smith
+    given-names: Arfon M.
+  - family-names: Katz
+    given-names: Daniel S.
+  - family-names: Niemeyer
+    given-names: Kyle E.
+  - name: "FORCE11 Software Citation Working Group"
+title: "Software citation principles"
```
Comment (arfon) [inline test/files/smith-et-al.cff:11]: @hainesr - I added these fields to actually make this CFF file valid.

Comment (hainesr) [reply test/files/smith-et-al.cff:11]: Great, thanks!

**Review by hainesr — APPROVED** [review]
Are we good to go with this now do you think?

**Comment by arfon** [comment]
> Are we good to go with this now do you think?
 
Yep, I think so!

**Comment by hainesr** [comment]
Ace. Thanks for kicking off this PR @arfon. I'm a lot more confident we've got the outputs covered better now! 🚀

**Comment by tgamblin** [comment]
How long til this hits GitHub?

**Comment by tgamblin** [comment]
Thanks everyone! How long do features here generally take to make it onto GitHub? Curious when we can start using preferred-citation in `CITATION.cff` files!

**Comment by arfon** [comment]
> Thanks everyone! How long do features here generally take to make it onto GitHub? Curious when we can start using preferred-citation in CITATION.cff files!

🤞 this week.

**Comment by tgamblin** [comment]
Thanks again -- this is a great addition to GitHub!

**Pull request merged**